task/yq/0.4/yq.yaml

```diff
@@ -0,0 +1,51 @@
+apiVersion: tekton.dev/v1beta1
+kind: Task
+metadata:
+  name: yq
+  labels:
+    app.kubernetes.io/version: "0.4"
+  annotations:
+    tekton.dev/pipelines.minVersion: "0.12.1"
+    tekton.dev/categories: Developer Tools
+    tekton.dev/tags: yq
+    tekton.dev/displayName: "YQ"
+    tekton.dev/platforms: "linux/amd64"
+spec:
+  description: >-
+    This task can be used to replace fields in YAML files. For example for altering helm charts on GitOps repos.
+  workspaces:
+    - name: source
+      description: A workspace that contains the file which needs to be altered.
+  params:
+    - name: SCRIPT
+      type: string
+      description: The yq script to execute. Can be multiple lines for complex tasks.
+    - name: image
+      type: string
+      description: The yq image to use.
+      default: docker.io/mikefarah/yq:4.16.2@sha256:0d4f6e27bdcac7316f635acd524ab0eecc4ad50834b54d10322268650c7712cb
+    - name: files
+      type: array
+      description: (deprecated, use SCRIPT instead) A list of files to execute the expression on. Needs to be relative to the source workspace.
+      default: []
+    - name: expression
+      type: string
+      description: (deprecated, use SCRIPT instead) The yq expression to apply. Can be used to replace yaml fields.
+      default: ""
```

Comment:
@vdemeester @piyush-garg how does this idea of deprecating the params look like? Removing them directly will break the users. WDYT?

Reply:
If it's a new version, we can definitely "break" the user. As commented elsewhere, we do not make any "backward compatible" guarantees from a version to another in the catalog. So from my point of view, removing it directly works for me.

Reply:
Should I change that? Or is it also fine as is?